src/components/SupportRequestModal.js

```diff
@@ -14,91 +13,96 @@ import validate from 'validate.js'
-    const { contactEmail, email } = props.authState.profile
+const SupportRequestModal =
+  Utils.connectAtom(authStore, 'authState')(class SupportRequestModal extends Component {
+    constructor(props) {
+      super(props)
+      const { contactEmail, email } = props.authState.profile
 
-    this.state = {
-      subject: '',
-      description: '',
-      type: 'question',
-      email: contactEmail || email
+      this.state = {
+        subject: '',
+        description: '',
+        type: 'question',
+        email: contactEmail || email || '',
+        nameEntered: ''
+      }
     }
-  }
 
-  render() {
-    const { onDismiss, authState: { profile: { firstName } } } = this.props
-    const { submitting, submitError, subject, description, type, email } = this.state
-    const greetUser = firstName === 'N/A' ? `?` : `, ${firstName}?`
-    const errors = validate({ email, description, subject }, constraints)
+    render() {
+      const { onDismiss, authState: { isSignedIn, profile: { firstName } } } = this.props
+      const { submitting, submitError, subject, description, type, email, nameEntered } = this.state
+      const greetUser = isSignedIn ? `, ${firstName}` : ``
+      const errors = validate({ email, description, subject, name: isSignedIn ? firstName : nameEntered }, constraints)
 
-    return h(Modal, {
-      onDismiss,
-      title: 'Contact Us',
-      okButton: buttonPrimary({
-        disabled: errors,
-        tooltip: Utils.summarizeErrors(errors),
-        onClick: () => this.submit()
-      }, ['SEND'])
-    }, [
-      Forms.requiredFormLabel('Type'),
-      h(Select, {
-        isMulti: false,
-        value: type,
-        onChange: ({ value }) => this.setState({ type: value }),
-        options: [{ value: 'question', label: 'Question' }, { value: 'bug', label: 'Bug' }, { value: 'feature_request', label: 'Feature Request' }]
-      }),
-      Forms.requiredFormLabel(`How can we help you${greetUser}`),
-      textInput({
-        style: { borderBottomLeftRadius: 0, borderBottomRightRadius: 0, borderBottomStyle: 'dashed' },
-        placeholder: 'Enter a subject',
-        autoFocus: true,
-        value: subject,
-        onChange: e => this.setState({ subject: e.target.value })
-      }),
-      h(TextArea, {
-        style: { height: 200, borderTopLeftRadius: 0, borderTopRightRadius: 0, borderTopStyle: 'dashed' },
-        placeholder: 'Enter a description',
-        value: description,
-        onChange: e => this.setState({ description: e.target.value })
-      }),
-      Forms.requiredFormLabel('Contact email'),
-      validatedInput({
-        inputProps: {
+      return h(Modal, {
+        onDismiss,
+        title: 'Contact Us',
+        okButton: buttonPrimary({
+          disabled: errors,
+          tooltip: Utils.summarizeErrors(errors),
+          onClick: () => this.submit()
+        }, ['SEND'])
+      }, [
+        !isSignedIn && Forms.requiredFormLabel('Name'),
+        !isSignedIn && textInput({
```

Comment:
Even though the logic driving these is simple, it would still be nice to drive home the idea that they come and go as a unit. We could wrap the two elements in a Fragment and then just have one conditional.